File: rb/features.rb
# ParkhausBasel SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ParkhausBaselFeatures
  def self.make_feature(name)
    case name
    when "base"
      ParkhausBaselBaseFeature.new
    when "ratelimit"
      ParkhausBaselRatelimitFeature.new
    when "retry"
      ParkhausBaselRetryFeature.new
    when "test"
      ParkhausBaselTestFeature.new
    when "timeout"
      ParkhausBaselTimeoutFeature.new
    else
      ParkhausBaselBaseFeature.new
    end
  end
end

File: py/features.py
# ParkhausBasel SDK feature factory

from feature.base_feature import ParkhausBaselBaseFeature
from feature.test_feature import ParkhausBaselTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ParkhausBaselBaseFeature(),
        "test": lambda: ParkhausBaselTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
